BUG: Test interactive default limit failing

This test is failing. xfailing for now, to make the CI in #2335, but we should fix the bug.
```
2020-08-18T11:50:03.5471755Z ____________________ TestInteractiveUse.test_default_limit _____________________
2020-08-18T11:50:03.5471849Z 
2020-08-18T11:50:03.5471963Z self = <ibis.expr.tests.test_interactive.TestInteractiveUse testMethod=test_default_limit>
2020-08-18T11:50:03.5472217Z 
2020-08-18T11:50:03.5472475Z         def test_default_limit(self):
2020-08-18T11:50:03.5473034Z             table = self.con.table('functional_alltypes')
2020-08-18T11:50:03.5473138Z     
2020-08-18T11:50:03.5473361Z             with config.option_context('interactive', True):
2020-08-18T11:50:03.5473469Z                 repr(table)
2020-08-18T11:50:03.5473562Z     
2020-08-18T11:50:03.5473653Z             expected = """\
2020-08-18T11:50:03.5474271Z     SELECT *
2020-08-18T11:50:03.5474366Z     FROM functional_alltypes
2020-08-18T11:50:03.5474458Z     LIMIT {0}""".format(
2020-08-18T11:50:03.5474543Z                 config.options.sql.default_limit
2020-08-18T11:50:03.5475140Z             )
2020-08-18T11:50:03.5475228Z     
2020-08-18T11:50:03.5475324Z >           assert self.con.executed_queries[0] == expected
2020-08-18T11:50:03.5475841Z E           AssertionError: assert 'SELECT *\nFR...onal_alltypes' == 'SELECT *\nFR...s\nLIMIT None'
2020-08-18T11:50:03.5475949Z E               SELECT *
2020-08-18T11:50:03.5476158Z E             - FROM functional_alltypes
2020-08-18T11:50:03.5476351Z E             ?                         -
2020-08-18T11:50:03.5476446Z E             + FROM functional_alltypes
2020-08-18T11:50:03.5476643Z E             - LIMIT None
2020-08-18T11:50:03.5476711Z 
2020-08-18T11:50:03.5476793Z ibis/expr/tests/test_interactive.py:64: AssertionError
```

--- a/ibis/tests/expr/test_interactive.py
+++ b/ibis/tests/expr/test_interactive.py
@@ -12,119 +12,117 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import unittest
-
 import pytest
 
 import ibis.config as config
 from ibis.tests.expr.mocks import MockBackend
 
 
-class TestInteractiveUse(unittest.TestCase):
-    def setUp(self):
-        self.con = MockBackend()
+@pytest.fixture
+def con():
+    return MockBackend()
 
-    def test_interactive_execute_on_repr(self):
-        table = self.con.table('functional_alltypes')
-        expr = table.bigint_col.sum()
-        with config.option_context('interactive', True):
-            repr(expr)
 
-        assert len(self.con.executed_queries) > 0
+def test_interactive_execute_on_repr(con):
+    table = con.table('functional_alltypes')
+    expr = table.bigint_col.sum()
+    with config.option_context('interactive', True):
+        repr(expr)
 
-    def test_repr_png_is_none_in_interactive(self):
-        table = self.con.table('functional_alltypes')
+    assert len(con.executed_queries) > 0
 
-        with config.option_context('interactive', True):
-            assert table._repr_png_() is None
 
-    def test_repr_png_is_not_none_in_not_interactive(self):
-        pytest.importorskip('ibis.expr.visualize')
+def test_repr_png_is_none_in_interactive(con):
+    table = con.table('functional_alltypes')
 
-        table = self.con.table('functional_alltypes')
+    with config.option_context('interactive', True):
+        assert table._repr_png_() is None
 
-        with config.option_context(
-            'interactive', False
-        ), config.option_context('graphviz_repr', True):
-            assert table._repr_png_() is not None
 
-    # XXX This test is failing in the OmniSciDB/Spark build, and working
-    # in the rest, even if does not seem to depend on the backend.
-    # For some reason in that build the statement does not contain
-    # the LIMIT. Xfailing with `strict=False` since in the other backends
-    # it does work. See #2337
-    @pytest.mark.xfail(
-        reason='Not obvious why this is failing for omnisci/spark, and this '
-        'was incorrectly skipped until now. Xfailing to restore the CI',
-        strict=False,
-    )
-    def test_default_limit(self):
-        table = self.con.table('functional_alltypes')
+def test_repr_png_is_not_none_in_not_interactive(con):
+    pytest.importorskip('ibis.expr.visualize')
 
-        with config.option_context('interactive', True):
-            repr(table)
+    table = con.table('functional_alltypes')
 
-        expected = """\
+    with config.option_context('interactive', False), config.option_context(
+        'graphviz_repr', True
+    ):
+        assert table._repr_png_() is not None
+
+
+def test_default_limit(con):
+    table = con.table('functional_alltypes')
+
+    with config.option_context('interactive', True):
+        repr(table)
+
+    expected = """\
 SELECT *
 FROM functional_alltypes
 LIMIT {}""".format(
-            config.options.sql.default_limit
-        )
+        config.options.sql.default_limit
+    )
 
-        assert self.con.executed_queries[0] == expected
+    assert con.executed_queries[0] == expected
 
-    def test_respect_set_limit(self):
-        table = self.con.table('functional_alltypes').limit(10)
 
-        with config.option_context('interactive', True):
-            repr(table)
+def test_respect_set_limit(con):
+    table = con.table('functional_alltypes').limit(10)
 
-        expected = """\
+    with config.option_context('interactive', True):
+        repr(table)
+
+    expected = """\
 SELECT *
 FROM functional_alltypes
 LIMIT 10"""
 
-        assert self.con.executed_queries[0] == expected
+    assert con.executed_queries[0] == expected
 
-    def test_disable_query_limit(self):
-        table = self.con.table('functional_alltypes')
 
-        with config.option_context('interactive', True):
-            with config.option_context('sql.default_limit', None):
-                repr(table)
+def test_disable_query_limit(con):
+    table = con.table('functional_alltypes')
 
-        expected = """\
+    with config.option_context('interactive', True):
+        with config.option_context('sql.default_limit', None):
+            repr(table)
+
+    expected = """\
 SELECT *
 FROM functional_alltypes"""
 
-        assert self.con.executed_queries[0] == expected
+    assert con.executed_queries[0] == expected
 
-    def test_interactive_non_compilable_repr_not_fail(self):
-        # #170
-        table = self.con.table('functional_alltypes')
 
-        expr = table.string_col.topk(3)
+def test_interactive_non_compilable_repr_not_fail(con):
+    # #170
+    table = con.table('functional_alltypes')
 
-        # it works!
-        with config.option_context('interactive', True):
-            repr(expr)
+    expr = table.string_col.topk(3)
 
-    def test_histogram_repr_no_query_execute(self):
-        t = self.con.table('functional_alltypes')
-        tier = t.double_col.histogram(10).name('bucket')
-        expr = t.group_by(tier).size()
-        with config.option_context('interactive', True):
-            expr._repr()
-        assert self.con.executed_queries == []
+    # it works!
+    with config.option_context('interactive', True):
+        repr(expr)
 
-    def test_compile_no_execute(self):
-        t = self.con.table('functional_alltypes')
-        t.double_col.sum().compile()
-        assert self.con.executed_queries == []
 
-    def test_isin_rule_supressed_exception_repr_not_fail(self):
-        with config.option_context('interactive', True):
-            t = self.con.table('functional_alltypes')
-            bool_clause = t['string_col'].notin(['1', '4', '7'])
-            expr = t[bool_clause]['string_col'].value_counts()
-            repr(expr)
+def test_histogram_repr_no_query_execute(con):
+    t = con.table('functional_alltypes')
+    tier = t.double_col.histogram(10).name('bucket')
+    expr = t.group_by(tier).size()
+    with config.option_context('interactive', True):
+        expr._repr()
+    assert con.executed_queries == []
+
+
+def test_compile_no_execute(con):
+    t = con.table('functional_alltypes')
+    t.double_col.sum().compile()
+    assert con.executed_queries == []
+
+
+def test_isin_rule_supressed_exception_repr_not_fail(con):
+    with config.option_context('interactive', True):
+        t = con.table('functional_alltypes')
+        bool_clause = t['string_col'].notin(['1', '4', '7'])
+        expr = t[bool_clause]['string_col'].value_counts()
+        repr(expr)

